doc/specs/#182 - Support for installation of portable standalone apps.md

```diff
@@ -0,0 +1,196 @@
+
+The corresponding settings are "PortablePackageUserRoot" and "PortablePackageUserRoot".
+
+>Note: The "packageIdentifier" will be used to generate subdirectories for these binaries to be installed to.
+
+Using Gitlab.gitlab runner as an example would result in the .exe being placed in "%LOCALAPPDATA%/WinGet_Packages/User/Gitlab.gitlab-runner/"
+If a user has configured a path to be used for portable applications, that path should be honored. The user should also be able to specify if they want all portable packages placed in the same directory, or in a directory per package.
+
+A related setting for the uninstall scenario will be able to specify the default behavior for either "--clean" or "--purge".
+
+### Install
+
+When the portable application is being installed by the Windows Package Manager, an entry will be created in Windows Apps & Features so the user will be able to see that the application is installed.
+
+The "AppPath" registry subkey will identify the path to the application so the user will be able to execute the program from any path via command prompt.
+
+The data from “AppsAndFeaturesEntry” in the manifest will be used to specify the name of the package in Windows Apps & Features.  Constraints for manifests with portable packages are covered below in the section on validating manifests. If the “AppsAndFeaturesEntry” doesn’t have a “DisplayName” value, then we will use the “PackageName”.
+The Product Code will be populated with the "PackageIdentifier" if it is not specified in the “AppsAndFeaturesEntry”.
+
+No shortcuts or icons will be created for the package.
+
+>Note: Some portable programs may require dependencies. Dependencies are not covered in this specification and are not yet supported as of the writing of this specification.
+
+### Upgrade
+
+The package is upgraded in the same path as the installed version. The first step the Windows Package Manager will perform is an uninstall of the previous version of the package. If the file is in use, the user will be informed the package is running so they can shut it down. Optionally, the user may specify "--force" to forcefully shut the application down for upgrade.
+
+A new Windows Apps & Features entry is created to correctly report the upgraded version for future potential upgrades.
+
+There will be no support for installing multiple “side by side” versions of portable packages.
+
+### Uninstall
+
+The executable should be removed along with the entry in Apps & Features. The user should also be able to uninstall the package from Apps & Features.
+
+>Note: the default behavior for uninstalling a portable application with Windows Apps & Features will be to execute uninstall with "--purge" so any files created by the portable application will be removed if they are located in the portable applications directory.
```

Comment:
I disagree on the default here; we should not purge.  I'm not unmovable, but generally it seems like if we already inform them of the extra files, they can act.

It also makes installing to a non-empty directory very dangerous for the future.  I think generally maybe purge shouldn't work if a custom, existing or non-empty location was chosen to prevent accidental nukage.

Reply:
The concern is if we are calling the uninstall from Apps & Features, the user wouldn't get informed. If we had a way to inform the user, then I'd agree. I just don't know what's possible here.